tests/mountingOptions/teleportTarget.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { mount } from '../../src'
+import ComponentWithTeleport from '../components/ComponentWithTeleport.vue'
+
+describe('mountingOptions.teleportTarget', () => {
+  const nestedTeleportTarget = document.createElement('div')
+  nestedTeleportTarget.id = 'teleport-target'
+  document.body.appendChild(nestedTeleportTarget)
+
+  it('should find the html after teleporting element', async () => {
+    const wrapper = mount(ComponentWithTeleport, {
+      slots: {
+        default: '<div>default slot content</div>',
+        ['nested-teleported-slot']: '<div>nested teleport-content</div>'
+      }
+    })
+
+    expect(wrapper.html()).not.toContain('slot content')
+    // @ts-expect-error
+    await wrapper.vm.toggleVisibility()
+    expect(wrapper.html()).toContain('slot content')
```

Comment:
It would makes sense that the test makes sure the wrapper html doesn't contain `'nested teleport-content'`

Reply:
I think this can be up for discussion, perhaps resolve all teleports or just one level?

If we're happy with this approach, I think what we need is:
 -  a consensus of what mount option(s) we need
 - loads of tests, covering all that we can think of

Then this PR can be progressed further

Reply:
Everything is up for discussion but if the rendered component doesn't contain the HTML, it makes sense for the tested component not to as well

Reply:
But that's the point of this PR to resolve the elements that are not in the component but has been teleported out.

Or are you saying only resolve 1 level of teleports?

Reply:
I think @posva is saying that if something does not appear in the html for a component (like wrapper.html()) it should not be findable with `find` etc.

I think we will need to reconsider - using `wrapper[method]` for something *not* in the wrapper's markup feels a little weird.

Reply:
I think an argument can be made that you're still interacting with content that was defined in the component. But I see your point given that the element in the wrapper doesn't actually contain the expected markup anymore.

Reply:
I'm saying both:

- `wrapper.html()` should not include html from teleported components
- `wrapper.find()` should not include elements from teleported components _by default_

Because that's closer to reality and therefor easier for tests